### ISSUE TITLE
Display the source that served up the packages

### ISSUE THREAD

**Issue by zjrunner** [NuGet/Home#1835]
Diagnostic logging shows the download and query url, but that isn't always easily mappable to the Source endpoint.  Here is the verbose output for project.json for a simple package and two source feeds -- they are both under the same account but different endpoints, and they get translated to a guid at some point and so aren't mappable back through console output:

```
NotFound https://corext.pkgs.visualstudio.com/DefaultCollection/_packaging/ed6df034-e254-4773-a194-71e620ed0301/nuget/v3/flat2/corext.extras/index.json 1023ms

OK https://corext.pkgs.visualstudio.com/DefaultCollection/_packaging/cb1c5ed0-3349-4f85-b393-2c8d0864b9b4/nuget/v3/flat2/corext.extras/1.0.2/corext.extras.1.0.2.nupkg 2695ms
```

_My order of preference:_
- new optional -downloadMetrics <file> that was simply a new optional output file with an array of Id, Version, Source json objects.  Could put download times and other metadata in this as well.
- entry in project.lock.json
- console diagnostic output

I could potentially get the last option in quickly by myself if approved.  I'd love to see the first option.

**Comment by yishaigalatzer** [comment]
TBH would have been a lot nicer if our friends from VSO made the urls directly useful and map to a feed address.

I'm fine with the 1st or the 3rd option, if you have the time to get in the 3rd right away go for it (and lets write it in a way that can easily be extracted from the log). The download metric thing is cool, but I don't see assigning someone to it right away (could be a cool summer intern project this year).

**Comment by zjrunner** [comment]
Update - I did find that I can back-map guid to name via another rest endpoint.  I'll see how easy it is for me to get the simple console solution into nuget.exe - if it means plumbing new data through lots of layers it may be better to hope for a downloadMetrics feature and fall back to the extra lookup from my side.

**Comment by yishaigalatzer** [comment]
@zjrunner waiting for your decision on this, we can pick it up after the first of the year

**Comment by zjrunner** [comment]
It didn't look that difficult for v3, but unfortunately it seems the packages.config flow still goes through nuget.core and that is the path I need currently (and upgrading to packages.json has other complexities at the moment).  I'm not as interested in altering nuget.core, especially since I've heard you are going after moving off of it and onto the published v2 protocol packages.  Once that happens I think the simple console-output solution is easily done and more inclusive of both flows.

**Comment by yishaigalatzer** [comment]
That's a good assessment. We don't have a concrete plan today, but I hope that we can have a better understanding by the end of the week.

@emgarten

**Comment by rrelyea** [comment]
4.0 RC - VSIX no longer has dependency on NuGet.Core
4.0 RC3 - working on removing NuGet.Core completely from NuGet.exe as well.
NuGet.exe may have remove NuGet.Core from this code path prior to that 4.0rc3 release...somebody should try 3.5.

**Comment by emgarten** [comment]
@zjrunner is this still needed?

**Comment by zjrunner** [comment]
The console change got added back in 3.5-ish timeframe and I was able to parse out of that.  I still really believe in a -downloadMetrics option that would allow callers to ingest telemetry to their own systems that included times and sources but in all honesty I wouldn't be consuming it right away so it is lower priority.  I'm more interested in a lockfile which would give a full summary of all packages used and all feeds than only download metrics.
